### PR TITLE
Draft PR to have FeedIterator<T> use FeedIterator

### LINKED ISSUE #413: Item query iterator should call item query stream iterator
To avoid causing different behavior between the two iterators the typed iterator should just call the stream iterator to do the conversion.

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description
This is a DRAFT PR to get feedback on the design. I only updated Items to support the FeedIterator changes. I will update the rest after I get feedback.

The feed iterators inherit from two different base abstract interfaces. In all the point operations the typed API just calls the stream implementation. This updates the FeedIterator<T> to use the FeedIterator for streams instead of having a separate implementation. 

Pros:

- Follows the rest of the SDK design
- Ensures that the typed tests also test the Stream APIs. This prevents duplicate tests and also ensure that they have the same behavior.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Closing issues

Put closes #413

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Container/CosmosContainerCore.Items.cs [hunk 1]
```diff
@@ -464,24 +471,17 @@ public override IOrderedQueryable<T> CreateItemQuery<T>(object partitionKey = nu
...
         {
-            CosmosQueryExecutionContext cosmosQueryExecution = (CosmosQueryExecutionContext)state;
-            QueryResponse queryResponse = await cosmosQueryExecution.ExecuteNextAsync(cancellationToken);
-            queryResponse.EnsureSuccessStatusCode();
+            cosmosResponseMessage.EnsureSuccessStatusCode();
 
             return QueryResponse<T>.CreateResponse<T>(
-                cosmosQueryResponse: queryResponse,
-                jsonSerializer: this.ClientContext.CosmosSerializer,
-                hasMoreResults: !cosmosQueryExecution.IsDone);
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Container/CosmosContainerCore.Items.cs]: Is this expected?

File: Microsoft.Azure.Cosmos/src/Resource/Container/CosmosContainerCore.Items.cs [hunk 2]
```diff
@@ -646,11 +646,12 @@ private object CosmosElementToPartitionKeyObject(CosmosElement cosmosElement)
                     QueryRequestOptions.FillContinuationToken(request, continuationToken);
                     QueryRequestOptions.FillMaxItemCount(request, maxItemCount);
                 },
-                responseCreator: response => response,
                 cosmosContainerCore: this,
                 partitionKey: null,
                 streamPayload: null,
                 cancellationToken: cancellationToken);
+
+            return (true, response.Headers.Continuation, response);
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Container/CosmosContainerCore.Items.cs]: Why are we not using a more declarative way for the response? I would rather have one extra internal class, than in the future having to track down what (for example) true means as first parameter in the response.

File: Microsoft.Azure.Cosmos/src/Resource/QueryResponses/FeedIteratorCore.cs [hunk 3]
```diff
@@ -147,12 +130,8 @@ internal class FeedIteratorCore<T> : FeedIterator<T>
         /// <returns>A query response from cosmos service</returns>
         public override async Task<FeedResponse<T>> FetchNextSetAsync(CancellationToken cancellationToken = default(CancellationToken))
         {
-            cancellationToken.ThrowIfCancellationRequested();
-
-            FeedResponse<T> response = await this.nextResultSetDelegate(this.MaxItemCount, this.continuationToken, this.queryOptions, this.state, cancellationToken);
-            this.HasMoreResults = response.HasMoreResults;
-            this.continuationToken = response.InternalContinuationToken;
-            return response;
+            CosmosResponseMessage response = await this.feedIteratorStreamCore.FetchNextSetAsync(cancellationToken);
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/QueryResponses/FeedIteratorCore.cs]: Is `feedIteratorStreamCore.FetchNextSetAsync` calling `ThrowIfCancellationRequested`?

Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Resource/QueryResponses/FeedIteratorCore.cs]: Even if it is we should probably call it at every layer that takes a cancellation token.

**Comment by j82w** [comment]
No longer needed. https://github.com/Azure/azure-cosmos-dotnet-v3/pull/481

**Pull request closed without merging**